src/hotspot/cpu/x86/x86_32.ad

```diff
@@ -1383,24 +1383,12 @@ void MachUEPNode::format( PhaseRegAlloc *ra_, outputStream* st ) const {
 
 void MachUEPNode::emit(CodeBuffer &cbuf, PhaseRegAlloc *ra_) const {
   MacroAssembler masm(&cbuf);
-#ifdef ASSERT
-  uint insts_size = cbuf.insts_size();
-#endif
-  masm.cmpptr(rax, Address(rcx, oopDesc::klass_offset_in_bytes()));
-  masm.jump_cc(Assembler::notEqual,
-               RuntimeAddress(SharedRuntime::get_ic_miss_stub()));
-  /* WARNING these NOPs are critical so that verified entry point is properly
-     aligned for patching by NativeJump::patch_verified_entry() */
-  int nops_cnt = 2;
-  if( !OptoBreakpoint ) // Leave space for int3
-     nops_cnt += 1;
-  masm.nop(nops_cnt);
-
-  assert(cbuf.insts_size() - insts_size == size(ra_), "checking code size of inline cache node");
+  masm.ic_check(CodeEntryAlignment);
```

Comment:
I think there may still be a problem with OptoBreakpoint messing up the alignment.  I'll try to reproduce it on x64.